code/_onclick/hud/spell_screen_objects.dm

```diff
@@ -169,7 +169,27 @@
 
 	var/icon/last_charged_icon
 	var/channeling_image
-
+	
+/obj/screen/spell/MouseEntered(location,control,params)
+	if(!spell)
+		return
+	var/dat = ""
+	if(spell.charge_type & Sp_RECHARGE)
+		dat += "<br>Cooldown: [spell.charge_max/10] second\s"
+	if(spell.charge_type & Sp_CHARGES)
+		dat += "<br>Has [spell.charge_counter] charge\s left"
+	if(spell.charge_type & Sp_HOLDVAR)
+		dat += "<br>Requires [spell.holder_var_amount] [spell.holder_var_type]"
+	switch(spell.range)
+		if(1)
+			dat += "<br>Range: Adjacency"
+		if(2 to INFINITY)
```

Comment:
Range can also be -1 , which is the user only, and -2, which is all mobs. 
Those should really be defines.